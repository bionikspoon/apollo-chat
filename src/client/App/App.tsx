import * as React from 'react'
import './App.css'

import { ApolloProvider } from 'react-apollo'
import { client } from '../apollo'
import ChatBox from '../ChatBox'
import Messages from '../Messages'

export default function App() {
  return (
    <ApolloProvider client={client}>
      <Messages />
      <ChatBox />
    </ApolloProvider>
  )
}
