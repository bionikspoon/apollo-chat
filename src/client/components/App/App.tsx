import React from 'react'
import { ApolloProvider } from 'react-apollo'
import Loadable from 'react-loadable'
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom'
import { client } from '../../setupApollo'
import FullPageLoading from '../FullPageLoading'

const AsyncHome = Loadable({
  loader: () => import('../../pages/Home'),
  loading: FullPageLoading,
})

const AsyncChat = Loadable({
  loader: () => import('../../pages/Chat'),
  loading: FullPageLoading,
})

export default function App() {
  return (
    <ApolloProvider client={client}>
      <BrowserRouter>
        <Switch>
          <Route exact={true} path="/" component={AsyncHome} />
          <Route path="/chat" component={AsyncChat} />
          {process.env.NODE_ENV !== 'production' ? (
            <Route path="/loading" component={FullPageLoading} />
          ) : null}
          <Redirect to="/" />
        </Switch>
      </BrowserRouter>
    </ApolloProvider>
  )
}
