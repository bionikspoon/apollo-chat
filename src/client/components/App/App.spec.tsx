import { shallow } from 'enzyme'
import React from 'react'
import App from '.'

test('it matches snapshot', () => {
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
})
