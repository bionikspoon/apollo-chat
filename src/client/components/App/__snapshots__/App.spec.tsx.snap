// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<ApolloProvider
  client={
    DefaultClient {
      "cache": InMemoryCache {
        "addTypename": true,
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "storeFactory": [Function],
        },
        "data": ObjectCache {
          "data": Object {
            "$ROOT_QUERY.user": Object {
              "__typename": "ClientUser",
              "color": "primary",
              "username": "User",
            },
            "ROOT_QUERY": Object {
              "user": Object {
                "generated": true,
                "id": "$ROOT_QUERY.user",
                "type": "id",
                "typename": "ClientUser",
              },
            },
          },
        },
        "optimistic": Array [],
        "silenceBroadcast": false,
        "typenameDocumentCache": WeakMap {},
        "watches": Array [],
      },
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "ssrMode": false,
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {
              "$ROOT_QUERY.user": Object {
                "__typename": "ClientUser",
                "color": "primary",
                "username": "User",
              },
              "ROOT_QUERY": Object {
                "user": Object {
                  "generated": true,
                  "id": "$ROOT_QUERY.user",
                  "type": "id",
                  "typename": "ClientUser",
                },
              },
            },
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "typenameDocumentCache": WeakMap {},
          "watches": Array [],
        },
      },
      "version": "2.4.2",
      "watchQuery": [Function],
    }
  }
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/chat"
      />
      <Route
        component={[Function]}
        path="/loading"
      />
      <Redirect
        push={false}
        to="/"
      />
    </Switch>
  </BrowserRouter>
</ApolloProvider>
`;
